sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapp/AADWebSecurityConfigurerAdapter.java

```diff
@@ -51,7 +50,8 @@ protected void configure(HttpSecurity http) throws Exception {
                 .and()
             .logout()
                 .logoutSuccessHandler(oidcLogoutSuccessHandler())
-                .and();
+                .and()
+            .addFilterBefore(handleConditionalAccessFilter(), WebAsyncManagerIntegrationFilter.class);
```

Comment:
Any specific reason about `WebAsyncManagerIntegrationFilter`?

Reply:
When we need add a filter, spring security requires us to specify the [registration order,](https://github.com/spring-projects/spring-security/blob/master/config/src/main/java/org/springframework/security/config/annotation/web/builders/FilterComparator.java#L54).

I didn't find that it should be added after a specific filter(it works after all filters), so I added our custom filter to the first(`WebAsyncManagerIntegrationFilter` is the first filter in filterChain).

Reply:
Please add integration test for conditional access. It's OK to do it after this PR merged.